### DISCUSSION TITLE
Slinger connects to other slingbox when the chosen slingbox is not available

### DISCUSSION THREAD

**Discussion by robosmith** [GerryDazoo/Slinger#615] [General]
We have Slinger working flawlessly on 3 different HD-Pros, but when one of them has disconnected due to periodic need for reboot/unbricking, slinger will sometimes connect to one of the other HD-Pros instead.

I find this odd and wonder why it will connect to an HD-Pro which is not at the address requested. That does not seem to happen when the requested HD-Pro is turned off. Also, once slinger connects to the alternate HD-Pro, it wants to keep connecting to it even when the requested HD-Pro is back online. Pi 4 then needs to be rebooted to get back to normal connecting.

Perhaps this is due to code which searches for an available slingbox when the requested slingbox fails to connect?
Strange that it didn't happen when the requested HD-Pro is powered off. Though I haven't done rigorous testing to verify that is always the case.

**Comment by GerryDazoo** [comment]
Yes, could be. If the server can't find ANY slingboxes at their defined IP addresses it will send out a broadcast message and listen for a response, if it gets one and only one it will use that. Should be clear from the logs what's going on. If it finds more than one it gives up looking because it doesn't know which one to use. 
You don't need to reboot the Pi just restart the server and it will try finding the available slingboxes again.
Make sure you have the correct IP address for the second Slingbox configured.  Again, this should be printed out in the logs when it scans for the Slingboxes.
Are you using the  Unified_config.ini format to define your slingboxes?
Please also note that this scanning is only done when the server starts up. If both Sbs are online at the time you should not have any issues. Again, check the server output to verify the IP addresses. They may not be what you think they are if they are using DHCP to get their IP addresses. You might want to consider defining static IPs for them.
Are you using the

**Reply by robosmith** [reply]
Well, we commented out the finder id # for the other slingbox, but slinger is still connecting to it when the addressed slingbox is bricked.
What is the other way slinger would find it?

**Reply by GerryDazoo** [reply]
The finderid is not used by Slinger. It's printed out for information purposes only. Just make sure you have the proper IP and port numbers configured for the boxes and restart the server.

**Reply by robosmith** [reply]
@GerryDazoo Both HD-Pros are configured with static IPs and the same port number, but different names. I find it strange that Slinger only connects to the wrong HD-Pro when the addressed one is bricked. Usually it takes a while with a spinning circle before connecting like it is searching for something. If the addressed HD-Pro is working, Slinger finds it right away. If the addressed HD-Pro is powered off, Singer never connects.

**Comment by sp222222** [comment]
I have four slingboxes setup on my server and I also see this on occasion. The stream will stop on one and go to the next highest port on my list (which in this case goes from xx55 to xx60). I do not use the unified config and I think I ended up going back a version of sling code to get my 500 to behave when it's powered off and it defaults to SD channel and to also keep this "next avail sling" glitch.
